system/Helpers/url_helper.php

```diff
@@ -492,23 +492,29 @@ function auto_link($str, $type = 'both', $popup = FALSE): string
 {
 
 	/**
-	 * Prep URL
-	 *
-	 * Simply adds the http:// part if no scheme is included
+	 * Prep URL - Simply adds the http:// part if no scheme is included.
 	 *
+	 * Formerly used URI, but that does not play nicely with URIs missing
+	 * the scheme.
+	 * 
 	 * @param	string	the URL
 	 * @return	string
 	 */
 	function prep_url($str = ''): string
 	{
-		$uri = new \CodeIgniter\HTTP\URI($str);
-
-		if (empty($uri->getScheme()))
-		{
-			$uri->setScheme('http');
-		}
-
-		return (string) $uri;
+		$pos = strpos('//', $str);
+		if ($pos === false)
+			$str = 'http://'.$str;
+
+//		$uri = new \CodeIgniter\HTTP\URI($str);
```

Comment:
What issue were you seeing when it was missing the scheme? The code here specifically checked for empty schemes and then added one in. I'm not opposed to this method, just curious if it highlights some deeper issue....

And, if you're going away from the URI version, we should probably go back to the full CI3 version: 

```
if ($str === 'http://' OR $str === '')
{
	return '';
}

$url = parse_url($str);

if ( ! $url OR ! isset($url['scheme']))
{
	return 'http://'.$str;
}

return $str;
```